apps/wallet-dashboard/hooks/usePersistedNetwork.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useIotaClientContext } from '@iota/dapp-kit';
+import { NetworkConfiguration } from '@iota/iota-sdk/client';
+import { useLocalStorage } from '@iota/core';
+import toast from 'react-hot-toast';
+import { useEffect } from 'react';
+
+export function usePersistedNetwork() {
+    const clientContext = useIotaClientContext();
+    const activeNetwork = clientContext.network;
+
+    const LOCAL_STORAGE_KEY = 'network_iota-dashboard';
+
+    const [persistedNetwork, setPersistedNetwork] = useLocalStorage<string>(
+        LOCAL_STORAGE_KEY,
+        activeNetwork,
+    );
```

Comment:
there are some errors in the terminal with this hook
`localStorage error ReferenceError: window is not defined`